internal/controller/vrg_volrep.go

```diff
@@ -1408,11 +1415,65 @@ func (v *VRGInstance) checkVRStatus(pvc *corev1.PersistentVolumeClaim, volRep *v
 
 // validateVRStatus validates if the VolumeReplication resource has the desired status for the
 // current generation and returns true if so, false otherwise
-//   - When replication state is Primary, only Completed condition is checked.
-//   - When replication state is Secondary, all 3 conditions for Completed/Degraded/Resyncing is
-//     checked and ensured healthy.
+//   - When replication state is Primary, Validated and Completed conditions are checked.
+//   - When replication state is Secondary, Completed, Degraded and Resyncing conditions are checked and
+//     ensured healthy.
 func (v *VRGInstance) validateVRStatus(pvc *corev1.PersistentVolumeClaim, volRep *volrep.VolumeReplication,
 	state ramendrv1alpha1.ReplicationState,
+) bool {
+	// Check validated for primary during VRG deletion.
+	if state == ramendrv1alpha1.Primary && rmnutil.ResourceIsDeleted(v.instance) {
+		validated, ok := v.validateVRValidatedStatus(volRep)
```

Comment:
Here we can update the DataReady and Protected conditions since we have more specific error message. I remvoed this update since many tests failed, I think the issue is wrong validation in the test and it is not related to this update, but I focused on solving the main problem. We can improve this in the next step.